lib/ch/types.ex

```diff
@@ -441,8 +478,10 @@ defmodule Ch.Types do
   defp utf8_size(codepoint) when codepoint <= 0x10FFFF, do: 4
 
   defguardp is_alpha(a) when (a >= ?a and a <= ?z) or (a >= ?A and a <= ?Z)
+  defguardp is_numeric(char) when char >= ?0 and char <= ?9
 
-  defp decode_identifier(<<a, rest::bytes>>, len, original, stack, acc) when is_alpha(a) do
+  defp decode_identifier(<<a, rest::bytes>>, len, original, stack, acc)
+       when is_alpha(a) or is_numeric(a) do
```

Comment:
I'm assuming identifiers are alphanumeric for now. But it seems like ClickHouse allows more than that. These work:

```sql
select tuple(1 as a$, 2 as b);
select tuple(1 as a_, 2 as b);
```

I'll try to find out what is allowed in identifiers and update it later.